### ISSUE TITLE
Looking for issue good for first contribution

### ISSUE THREAD

**Issue by piccoloaiutante** [nodejs/build#495]
Hi @mhdawson,

we met at NodeInteractive and as I told you I would like to start contributing to this working group. Is there any issue that is a good starting point for someone that is doing his first contribution?

Thanks

**Comment by mhdawson** [comment]
Sorry been catching up after being away at Node Interactive.  Will try to come up with a good suggestion this week.

**Comment by Fishrock123** [comment]
ping @mhdawson

**Comment by mhdawson** [comment]
@Fishrock123 thanks for the reminder, will bring it up in WG meeting to day to see what kind of suggestions I can come up with.  @piccoloaiutante you may also want to join the meeting today to listen to get a feel with respect to the issues being discussed/worked.

**Comment by jbergstroem** [comment]
I guess it depends on what the interest might be. I can see a few things between this and the github-bot group.

**Comment by piccoloaiutante** [comment]
@mhdawson @jbergstroem i could join you in WG meeting in half hour so we can discuss about it

**Comment by jbergstroem** [comment]
@piccoloaiutante lets keep the discussion post the meeting. I can stick around a bit longer and I'm sure @mhdawson can too! You're still welcome to join in and listen to our meeting.

**Comment by piccoloaiutante** [comment]
@jbergstroem no problem, i'll follow it. Then i'm not sure that i'll stick around  after it since i'm based in italy, but anyway we can go on with this issue.

**Comment by joaocgreis** [comment]
Welcome @piccoloaiutante ! Here are some ideas:

- Adapt the [Windows Ansible scripts](https://github.com/nodejs/build/tree/master/setup/windows) to the [refactor by Johan](https://github.com/jbergstroem/build/tree/feature/refactor-the-world/ansible), integrating Windows as well as possible. Some things that would be nice to have:
  - Handling reboots, when updating Windows or installing Visual Studio.
  - Updating Windows before starting to install programs.
  - Generating `.rdp` and/or `.remmina` files (I have a rough script for Remmina that can serve as a starting point).
  - Handle Windows secrets better - passwords are used only for Windows and currently the file we have is not parsable, so we can decide for something better that Ansible can pick from the secrets (currently I have a custom script to generate `host_vars`).

- `vcbuild.bat` support for https://github.com/nodejs/build/issues/387 , porting [what is currently in the `Makefile`](https://github.com/nodejs/node/blob/4b312387ead4ba11146b28b8ac05ed385919c4af/Makefile#L263-L292) (introduced initially in https://github.com/nodejs/node/pull/4704 ).

**Comment by piccoloaiutante** [comment]
@joaocgreis I have a first draft of ansible script for supporting widows update and reboot. I tested it on windows 10 and windows server 2012. Can you have a quick look at it before I submit a PR for it just to see that i'm on track of what you're expecting?

https://gist.github.com/piccoloaiutante/d6f0766235b63d44defd3d35bc1887c7

**Comment by piccoloaiutante** [comment]
@joaocgreis i can get it Windows Server 2008R2 from msdn. Don't worry i'll be back with info about it. Thanks

**Comment by piccoloaiutante** [comment]
@joaocgreis i spent a lot of time trying to install power shell 3.0 on windows 2008 R2 so i could run the setup [script](https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1) but i didn't succeed. Is there anything particular that i should do for provisioning this OS?

**Comment by joaocgreis** [comment]
@piccoloaiutante I've used the script at http://docs.ansible.com/ansible/intro_windows.html#getting-to-powershell-3-0-or-higher before, it worked for me.

**Comment by joaocgreis** [comment]
I also had issues updating PowerShell in my local VM, the script didn't work. But the instructions at https://msdn.microsoft.com/en-us/powershell/scripting/setup/installing-windows-powershell#a-namebkmkinstallingonwindows7andwindowsserver2008r2ainstalling-windows-powershell-on-windows-7-and-windows-server-2008-r2 did it.

**Comment by joaocgreis** [comment]
As noted by @nschonni in https://github.com/nodejs/build/pull/750#issuecomment-307440878 , we don't have a `CONTRIBUTING.md`. We could add one pointing to https://github.com/nodejs/node/blob/master/CONTRIBUTING.md , with also a placeholder to mention any differences or additions. Furthermore, the templates (https://github.com/nodejs/node/tree/master/.github) would also be good to have. This could probably be a good first contribution.

**Comment by mhdawson** [comment]
@piccoloaiutante I think we can close this now as you are part of the WG :)
